widgets/calendar-short/api.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+const sortEvents = require('./../../lib/sort-events')
+
+const getDayKey = (datetime) => {
+  return datetime.clone().startOf('day');
+}
+
+module.exports = {
+
+  async getCalendarList({ homey, query }) {
+    const timeFormat = homey.app.variableMgmt.dateTimeFormat.time;
+    let events = homey.app.variableMgmt.calendars.reduce(
+      (allEvents, calendar) => {
+        // Create a copy of the calendar object without the 'events' property
+        const { events: _, ...calendarWithoutEvents } = calendar;
+
+        // For each event in the calendar, add the calendar reference
+        const eventsWithCalendar = calendar.events.map(event => ({
+          ...event,
+          calendar: calendarWithoutEvents // Attach the calendar reference
+        }));
+        return allEvents.concat(eventsWithCalendar);
+      },
+      []
+    );
+
+    // Ensure max X row is shown
+    events = sortEvents(events);
+    events = events.slice(0, query.max);
+
+    // Group events by day and serialize the data
```

Comment:
When you need to comment a code block like this, that's a sign you should take that code block out as a separate function and name it accordingly instead. That way it's cleaner and easier to read.